sdk/eventhub/Azure.Messaging.EventHubs/tests/Compatibility/TrackOneComparerTests.cs

```diff
@@ -198,13 +198,119 @@ public void IsEventDataEquivalentDetectsEqualEvents()
+        {
+            var body = new byte[] { 0x22, 0x44, 0x88 };
+            var offset = 27;
+            var trackTwoSystemProperties = new Dictionary<string, object>();
+
+            var trackTwoEvent = new EventData(
+                eventBody: (byte[])body.Clone(),
+                offset: offset,
+                systemProperties: trackTwoSystemProperties,
+                lastPartitionSequenceNumber: 9765551212,
+                lastPartitionOffset: 54321,
+                lastPartitionEnqueuedTime: DateTimeOffset.Parse("2015-10-27T00:00:00Z"));
+
+            trackTwoEvent.Properties["test"] = "same";
+
+            var trackOneEvent = new TrackOne.EventData((byte[])body.Clone());
+            trackOneEvent.Properties["test"] = trackTwoEvent.Properties["test"];
+            trackOneEvent.SystemProperties = new TrackOne.EventData.SystemPropertiesCollection();
+            trackOneEvent.SystemProperties[TrackOne.ClientConstants.OffsetName] = offset.ToString();
+            trackOneEvent.LastEnqueuedOffset = "1";
+            trackOneEvent.LastSequenceNumber = trackTwoEvent.LastPartitionSequenceNumber.Value;;
+            trackOneEvent.LastEnqueuedTime = trackTwoEvent.LastPartitionEnqueuedTime.Value.UtcDateTime;
+
+            Assert.That(TrackOneComparer.IsEventDataEquivalent(trackOneEvent, trackTwoEvent), Is.False);
+        }
+
+        /// <summary>
+        ///   Verifies functionality of the <see cref="IsEventDataEquivalent" /> test
+        ///   helper.
+        /// </summary>
+        ///
+        [Test]
+        public void IsEventDataEquivalentDetectsWhenLastEnqueuedTimeDiffers()
+        {
+            var body = new byte[] { 0x22, 0x44, 0x88 };
+            var offset = 27;
+            var trackTwoSystemProperties = new Dictionary<string, object>();
+
+            var trackTwoEvent = new EventData(
+                eventBody: (byte[])body.Clone(),
+                offset: offset,
+                systemProperties: trackTwoSystemProperties,
+                lastPartitionSequenceNumber: 9765551212,
+                lastPartitionOffset: 54321,
+                lastPartitionEnqueuedTime: DateTimeOffset.Parse("2015-10-27T00:00:00Z"));
+
+            trackTwoEvent.Properties["test"] = "same";
 
             var trackOneEvent = new TrackOne.EventData((byte[])body.Clone());
-            trackOneEvent.Properties["test"] = "same";
+            trackOneEvent.Properties["test"] = trackTwoEvent.Properties["test"];
+            trackOneEvent.SystemProperties = new TrackOne.EventData.SystemPropertiesCollection();
+            trackOneEvent.SystemProperties[TrackOne.ClientConstants.OffsetName] = offset.ToString();
+            trackOneEvent.LastEnqueuedOffset = trackTwoEvent.LastPartitionOffset.ToString();
+            trackOneEvent.LastSequenceNumber = trackTwoEvent.LastPartitionSequenceNumber.Value;;
+            trackOneEvent.LastEnqueuedTime = DateTime.Parse("2012-03-04T08:46:00Z").ToUniversalTime();
```

Comment:
nit: Having experienced weird DateTime.Parse errors (ie. yyyy-mm-dd? or yyyy-dd-mm?), I prefer DateTime.ParseExact("", "u") in this case so it'll blow up early. It should be able to infer the universal too because 'u' format is always UTC..

Reply:
huh. I totally forgot `ParseExact` exists.  I'm using the standard ISO-8601 format... but the parser ignores "Z" and makes it local time anyway.   That's a darn good idea, but I'll probably make a note and do it them all at once to keep them consistent.

Reply:
I feel like that's a bug if the standard specifies that it's UTC and it gives us local.. but anywayyysss. DateTime class itself is a giant black hole. 😆

Reply:
> If s contains no time zone information, the Kind property of the returned DateTime object is DateTimeKind.Unspecified. This behavior can be changed by using the DateTimeStyles.AssumeLocal flag, which returns a DateTime value whose Kind property is DateTimeKind.Local, or by using the DateTimeStyles.AssumeUniversal and DateTimeStyles.AdjustToUniversal flags, which returns a DateTime value whose Kind property is DateTimeKind.Utc. If s contains time zone information, the time is converted to local time, if necessary, and the Kind property of the returned DateTime object is set to DateTimeKind.Local. This behavior can be changed by using the DateTimeStyles.RoundtripKind flag to not convert Coordinated Universal Time (UTC) to a local time and to set the Kind property to DateTimeKind.Utc. 

Maybe DateTimeStyles.AssumeUniversal? But still.. Blegh
https://docs.microsoft.com/en-us/dotnet/api/system.datetime.parseexact?view=netframework-4.8#System_DateTime_ParseExact_System_String_System_String_System_IFormatProvider_System_Globalization_DateTimeStyles_

Reply:
This is why track two code is based on `DateTimeOffset`.    All the `DateTime` goes away when we remove the track one dependency for preview 4.  :)

Reply:
Tracking with [#7484](https://github.com/Azure/azure-sdk-for-net/issues/7484)